ui/app/pages/settings/networks-tab/network-form/network-form.component.js

```diff
@@ -294,7 +315,18 @@ export default class NetworkForm extends PureComponent {
 
     if (providerError || typeof endpointChainId !== 'string') {
       errorMessage = t('failedToFetchChainId')
-    } else if (chainId !== endpointChainId) {
+    } else if (parsedChainId !== endpointChainId) {
+      if (!formChainId.startsWith('0x')) {
```

Comment:
Why only do this try catch if `formChainId` does not start with '0x'?

Reply:
And why only do it if `parsedChainId !== endpointChainId`?

Reply:
I guess the answer to "And why only do it if parsedChainId !== endpointChainId?" is that `parsedChainId` is always a `0x` prefixed hex string, so if `parsedChainId === endpointChainId`, then we know `endpointChainId` is good. First question still stands though

Reply:
The logic in the function renamed to `validateChainIdOnChange` in 0be1e41 ensures that, by the time `onSubmit` is called, the `chainId` will be either a decimal or `0x`-prefixed hexadecimal string.

Line 318 in your comment checks that the endpoint returned the expected `chainId` value. If the user entered a decimal string into the form, we convert the endpoint `chainId` to decimal before rendering it in an error message.
So, this is purely for display purposes.

I added a comment explaining this in: b28aa9c